### PR TITLE
[draft] cron job logging for checkFreeTierBillingUsage

### PR DESCRIPTION
**Pull request by jaycarlton**
Added some logging around error conditions and general tidbits.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java [hunk 1]
```diff
@@ -43,26 +55,69 @@ public void checkFreeTierBillingUsage() {
...
+    if (queryResults.getTotalRows() < numWorkspaces) {
+      logger.warn("Expected %d rows but found %d", numWorkspaces, queryResults.getTotalRows());
+    }
+
+    Streams.stream(queryResults.getValues())
+        .filter(fv -> fv.get("id").isNull())
+        .forEach(
+            fv -> logger.error(String.format("Query result has no project ID: %s", fv.toString())));
 
     Map<User, Double> perUserSpend =
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java]: This entire stream should probably be written differently per my above comment - perhaps starting with the namespaceToCreator map and then joining against the query results.

File: api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java [hunk 2]
```diff
@@ -43,26 +55,69 @@ public void checkFreeTierBillingUsage() {
...
     Map<String, User> namespaceToCreator = workspaceDao.namespaceToCreator();
+    final int numWorkspaces = namespaceToCreator.keySet().size();
+    final long numUniqueUsers = namespaceToCreator.values().stream().distinct().count();
+    logger.info(
+        String.format("found %d workspaces with %d unique users", numWorkspaces, numUniqueUsers));
+
+    final TableResult queryResults = bigQueryService.executeQuery(queryConfig);
+    logger.info("Query retrieved %d results", queryResults.getTotalRows());
+
+    if (queryResults.getTotalRows() < numWorkspaces) {
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java]: OK, I'll try to document that.

File: api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java (hunk 1 above)
Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java]: Is it worth me doing that on master, or would you rather pick what's useful off this branch?

Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java]: Thanks - I can just pick things off this branch as needed

**Pull request closed without merging**